Lexplorer/Components/HomepageOverview.razor.cs

```diff
@@ -0,0 +1,119 @@
+            if (uniswapToken != null && pairCount == 0)
+            {
+                uniswapTokens = new List<UniswapToken>();
+                uniswapTokens!.Add(uniswapToken!);
+            }
+            else
+            {
+                uniswapTokens!.Add(uniswapToken!);
+            }
+            pairCount++;
+        }
+        StateHasChanged();
+    }
+
+    private decimal? priceOfToken(Token token)
+    {
+        //find token.address in uniswapTokens and get most recent price; if not available, return 1.0 so no conversion
+        return uniswapTokens?.Where(c => c.address == @token?.address).FirstOrDefault()?.data?.tokenDayDatas?[0].priceUSD;
+    }
+
+    private string? getVolumePair1(Pair pair, Boolean weekly = true)
+    {
+        if (pair == null) return null;
+        decimal? price = (pair.token1 == null) ? null : priceOfToken(pair.token1);
+        var volume = weekly
+            ? pair.weeklyEntities![0].tradedVolumeToken1Swap
+            : pair.dailyEntities![0].tradedVolumeToken1Swap;
+        if (price.HasValue)
+            return $"${@TokenAmountConverter.ToStringWithExponent(volume, pair.token1?.decimals ?? 0, price.Value)}";
+        else
+            return $"{pair.token1?.symbol} {@TokenAmountConverter.ToStringWithExponent(volume, pair.token1?.decimals ?? 0, 1)}";
+    }
+
+    private string? get24hPercentage(Pair pair)
+    {
+        if (pair == null) return null;
+        return (pair.dailyEntities![0].tradedVolumeToken1Swap * 7 / pair.weeklyEntities![0].tradedVolumeToken1Swap * 100).ToString("N2");
+    }
+
+    private void CalculateAverageBlockTime()
+    {
+        long transactionCount = 0;
+        long currentTime = DateTimeOffset.Now.ToUnixTimeSeconds();
+        List<long> timeBetweenBlocks = new List<long>();
+        foreach (var block in blockData!.data!.blocks!)
+        {
+            transactionCount += block.transactionCount;
+            timeBetweenBlocks.Add(currentTime - Int64.Parse(block.timestamp!));
+            currentTime = Int64.Parse(block.timestamp!);
+        }
+        averageTransactions = transactionCount / blockData.data.blocks.Count;
+        averageBlockTime = Math.Floor(timeBetweenBlocks.Average() / 60);
+    }
+
+    private void CalculateLastBlockSubmitted()
+    {
+        long currentTime = DateTimeOffset.Now.ToUnixTimeSeconds();
+        long timeSinceLastBlock = currentTime - Int64.Parse(blockData!.data!.blocks![0].timestamp!);
+        lastBlockSubmittedTime = Math.Floor((double)timeSinceLastBlock / 60);
+    }
+
+    private void GoToBlockOverviewPage()
+    {
+        string parameters = "blocks";
+        NavigationManager.NavigateTo(parameters);
+    }
+
+    private void GoToPairOverviewPage()
+    {
+        string parameters = "pairs";
+        NavigationManager.NavigateTo(parameters);
+    }
+
+    private void GoToTransactionOverviewPage()
+    {
+        string parameters = "transactions";
+        NavigationManager.NavigateTo(parameters);
+    }
+}
```

Comment:
What is to be gained from separating the code from the page here? I'm honestly asking - because personally I think that makes navigating between code and page just more cumbersome.

Reply:
Seperating the application logic from the presentation layer is to seperate concers and is viewed as good practise. Also gains the possibility to be tested without the need to prove UI. 
May be a preference thing and I also get your view as to navigating between layers. It's up to you and @fudgebucket27 :)

Reply:
See that's the point, of course you should separate application logic and presentation layer - but that code is part of the presentation layer! It cannot be tested separately. And if there is code in there that could - it should be moved out of it, exactly for that separation. That's what we did with TokenAmountConverter and LinkHelper.

Separating page and partial class now also makes us scramble commit history - as you can see by your commits, it's very difficult to judge whether and what has changed in addition to the separation.

Reply:
It's not always needed to be an extension class to be seperated. It's good practise to seperate code and maybe especially when dealing with html and C#. 
In respect to maintainability it would be down to preferences. In my view it's easier to maintain when seperated, but I can see other point of view of feeling different about this issue. :) 

And as a big point here, this is not my code and not my project so this is all up to you guys to decide what to do :)

The same goes to interfaces.